src/Language/PureScript/Docs/Convert.hs

```diff
@@ -108,17 +108,30 @@ insertValueTypesAndAdjustKinds env m =
   -- - `Constraint` (class declaration only)
   -- - `Type -> K` where `K` is an "uninteresting" kind
   isUninteresting
-    :: P.KindSignatureFor -> P.Type () -> Bool
+    :: P.KindSignatureFor -> Type' -> Bool
   isUninteresting keyword = \case
-    P.TypeApp _ t1 t2 | t1 == kindFunctionType ->
+    -- `Type -> ...`
+    P.TypeApp _ (P.TypeApp _ (P.TypeConstructor () Prim.Function) ty) t2 | isKindPrimType ty ->
```

Comment:
We need to account for the case where an application is wrapped in parens, like `(Type -> Type)`, otherwise we deem such kinds interesting and they show up in the generated docs.

Reply:
Meaning if we have something like this?
```purescript
data Foo :: (Type -> Type)
data Foo a = Foo a
```
which is the same as writing the uninteresting kind signature below:
```purescript
data Foo :: Type -> Type
data Foo a = Foo a
```

Reply:
Yeah exactly, `data Foo :: (Type -> Type)` should be treated like `data Foo :: Type -> Type` but the kind is currently documented.

Reply:
Wouldn't that mean accounting for various other situations, too? I'm starting to wonder if this is really worth doing and if we should instead say, "Yeah, just don't do that."

As examples of other things:
```purescript
-- or any uninteresting kind where the whole thing is wrapped in parens
data Foo1 :: (Type -> Type -> Type)
data Foo1 a b = Foo1 a b


-- or any uninteresting kind where the the tail part is wrapped in parens
data Foo2 :: Type -> (Type -> Type)
data Foo2 a b = Foo2 a b
```

Reply:
Yes those kinds should be uninteresting too. It should only be a matter of adding a  `P.ParensInType _ ty -> isUninteresting ty` case to `isUninteresting` though.

Reply:
Done and added tests for it. I haven't tested this locally.

Reply:
Does this cover `data Foo :: (->) Type Type`? Or `data Foo :: ((->) Type) Type`, or, for that matter, `data Foo :: ((((->)) Type)) Type`?

These are profoundly obnoxious edge cases, but if they cause problems, maybe the thing to do is just strip out all parens using a traversal before calling `isUninteresting`.

Reply:
Good question. I'll add a test for that.

Reply:
Latest commit adds these three tests, the latter two of which currently fail.

Reply:
Yeah, stripping those parenthesis out and then solving the problem with the original approach was much easier than trying to account for them at the same time.